### ISSUE TITLE
[BUG]: getting a lot of "cannot read properties of undefined" error messages

### ISSUE THREAD

**Issue by sabom2d** [edterbak/NodeRed_Heishamon_control#124]
Hi,

after solving the msg Error  I now get on both node-red installations "cannot read properties of undefined ()"errors:
12 Aug 17:06:26 - [info] [mqtt-broker:MQTT (x.x.x.x)] Verbindung zum Broker Node_Red_Client@mqtt://192.168.90.44:1884 aufgebaut
12 Aug 17:06:27 - [error] [function:Function - bucket 1 + 2 combined] TypeError: Cannot read properties of undefined (reading 'payload')
12 Aug 17:06:28 - [error] [function:Draw chart] TypeError: Cannot read properties of undefined (reading 'detection_method')
12 Aug 17:06:28 - [error] [function:Draw chart] TypeError: Cannot read properties of undefined (reading 'detection_method')
12 Aug 17:06:28 - [error] [function:Function RTC automation] TypeError: Cannot read properties of undefined (reading 'function_active_state')
12 Aug 17:06:31 - [error] [function:Draw chart] TypeError: Cannot read properties of undefined (reading 'detection_method')
12 Aug 17:06:32 - [error] [function:DHW SP Manager] TypeError: Cannot read properties of undefined (reading 'state')
12 Aug 17:06:32 - [error] [function:Draw chart] TypeError: Cannot read properties of undefined (reading 'detection_method')
12 Aug 17:06:32 - [error] [function:Function RTC automation] TypeError: Cannot read properties of undefined (reading 'function_active_state')
12 Aug 17:06:39 - [error] [function:Function RTC automation] TypeError: Cannot read properties of undefined (reading 'function_active_state')
12 Aug 17:06:47 - [error] [function:Function RTC automation] TypeError: Cannot read properties of undefined (reading 'function_active_state')
12 Aug 17:06:52 - [error] [function:Function RTC automation] TypeError: Cannot read properties of undefined (reading 'function_active_state')
12 Aug 17:07:22 - [error] [function:cee35b0a1e2da0ea] TypeError: Cannot read properties of undefined (reading 'function_active_state')

Anything I forgot to configure correctly?

Regards

**Comment by edterbak** [comment]
Sorry for the late reply. 
Can you give an update on the issue status?

I have never seen these errors from others. 
From the log part, it seems to come from everywhere. :(

**Comment by sabom2d** [comment]
Hi,     
  

  
Issue is still active.
  
Earlyest access to the system next friday.
  
Will give feedback then.
  

  
Regards
  

  

  

  

  
  
  
  
>

**Comment by sabom2d** [comment]
Hi, 

took a little longer to get back to you, but with good news.
I disabled the responsible function temporary for being able to use your tool, 
now that I activated it again, the issue seems gone, no Errors popping up any more:
![image](https://github.com/edterbak/NodeRed_Heishamon_control/assets/107671129/c9f1bf34-854f-47c4-8f38-ca1cc4e6baff)

Although I'm not sure why this solved the issue now, I think this was it.
Will see the next days , thanks again for your fine work.

regards
